pkg/operator/configobservation/auth/auth_serviceaccountissuer.go

```diff
@@ -71,6 +71,13 @@ func observedConfig(existingConfig map[string]interface{},
 		existingActiveIssuer = existingIssuers[0]
 	}
 
+	// if the issuer is not set, it is safe to assume it is being defaulted by the config override and we don't have
+	// to trigger revision rollout just to sync operator status with config.
+	if len(existingActiveIssuer) == 0 {
+		existingActiveIssuer = "https://kubernetes.default.svc"
+		existingIssuers = []string{existingActiveIssuer}
```

Comment:
A corner case: `len(existingIssuers) > 0`, `existingIssuer[0] = ""; existingIssuer[1] = "https://nice.issuer"`.

In that case you're rewriting `existingIssuers`, which is a field that you read from the observed config, with `{ <default_issuer> }`.

Perhaps looking for a first non-empty issuer in `existingIssuers` and then doing this would be safer. It shouldn't happen that an issuer in the observed slice would be empty, but you can never be too sure :eyes: